### PR TITLE
mes-8196-modalCheckFix

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Fixed icons displaying at the incorrect time

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
image
image
image

image
image

### CODE REVIEW

File: src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts [hunk 1]
```diff
@@ -33,6 +36,9 @@ export class ModalResultItemComponent {
 
   displayScore(score: number): string {
     const finalScore = (score || 0);
+    if (this.testState > 0) {
+      return this.adi3 ? (score || 0).toString() : `${finalScore}%`;
+    }
     return this.adi3 ? (score || '-').toString() : `${finalScore}%`;
   }
```
Comment (matthew2564) [inline src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts:43]:
```
  displayScore(score: number): string {
    const finalScore = (score || 0);
    return this.adi3 ? (typeof score === 'number' ? score || '-').toString() : `${finalScore}%`;
  }
```
Think we can change to the above and then not need to pass in a new input to component

File: src/app/pages/test-report/cat-adi-part3/test-report.cat-adi-part3.page.html [hunk 2]
```diff
@@ -73,12 +73,13 @@ <h3>
       </ion-col>
       <ion-col class="score-box" size="5">
         <p>
-          {{adi3AssessmentProvider.getTotalAssessmentScore(testDataADI3) === 0 ? '' : adi3AssessmentProvider.getTotalAssessmentScore(testDataADI3)}}
+          {{adi3AssessmentProvider.validateTestReport(
+                testDataADI3.lessonPlanning, testDataADI3.riskManagement, testDataADI3.teachingLearningStrategies) === 0
+                ? '' : adi3AssessmentProvider.getTotalAssessmentScore(testDataADI3)}}
```
Comment (matthew2564) [inline src/app/pages/test-report/cat-adi-part3/test-report.cat-adi-part3.page.html]: Is this necessary or been asked for?

File: src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts (hunk 1 above)
Comment (matthew2564) [inline src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts]: Not sure I'm a fan of this seeing as the entire return statement is duplicated to essentially swap the 0 & '-'. Does something like this work?
```
  displayScore(score: number): string {
       if (!this. adi3) {
           const finalScore = (score || 0);
           return `${finalScore}%`;
       }
       return (typeof score === 'number') ? score.toString() : '-';
  }
```

File: src/app/pages/test-report/cat-adi-part3/test-report.cat-adi-part3.page.html (hunk 2 above)
Comment (RLCorp) [reply src/app/pages/test-report/cat-adi-part3/test-report.cat-adi-part3.page.html]: changed in newest commit

File: src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/pages/test-report/cat-cpc/components/cpc-end-test-modal/components/modal-result-item/modal-result-item.ts]: changed in newest commit

**Review by matthew2564 — APPROVED** [review]

**Review by andrewsetterfield — APPROVED** [review]

**Pull request merged**